src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AvailabilityServiceTest.kt

```diff
@@ -218,4 +218,54 @@ class AvailabilityServiceTest {
       assertThat(alternatives).hasSize(0)
     }
   }
+
+  @Test
+  fun `Excludes appointments with the same videoBookingId as vlbIdToExclude`() {
+    // Request to exclude certain videoBookingId
+    val request = AvailabilityRequest(
+      bookingType = BookingType.COURT,
+      courtOrProbationCode = "TESTC",
+      prisonCode = MOORLAND,
+      date = LocalDate.now(),
+      mainAppointment = LocationAndInterval(
+        prisonLocKey = room1,
+        interval = Interval(start = LocalTime.of(10, 0), end = LocalTime.of(11, 0)),
+      ),
+      vlbIdToExclude = 2L,
+    )
+
+    val response = service.checkAvailability(request)
+
+    // We should exclude appointments with videoBookingId == 2L
+    assertThat(response).isNotNull
+    with(response) {
+      assertThat(availabilityOk).isTrue()
+      assertThat(alternatives).hasSize(0) // No conflicts after excluding
+    }
+  }
+
+  @Test
+  fun `Includes appointments not matching vlbIdToExclude`() {
+    // Request to exclude certain videoBookingId
+    val request = AvailabilityRequest(
+      bookingType = BookingType.COURT,
+      courtOrProbationCode = "TESTC",
+      prisonCode = MOORLAND,
+      date = LocalDate.now(),
+      mainAppointment = LocationAndInterval(
+        prisonLocKey = room1,
+        interval = Interval(start = LocalTime.of(10, 0), end = LocalTime.of(11, 0)),
+      ),
+      vlbIdToExclude = 5L,
+    )
+
+    val response = service.checkAvailability(request)
+
+    // We should not exclude appointments with videoBookingId == 2L (conflict exists)
```

Comment:
Not sure if I an reading this wrong but this says `2L` in the comment but looks like `5L` in the request?